content-security-policy/support/testharness-helper.sub.js

```diff
@@ -0,0 +1,47 @@
+const Host = {
+  SAME_ORIGIN: "same-origin",
+  CROSS_ORIGIN: "cross-origin",
+};
+
+const PolicyHeader = {
+  CSP: "echo-policy.py?policy=",
+  EMBEDDING_CSP: "echo-embedding-csp.py",
+} 
+
+function generateURL(host, path) {
+  var url = new URL("http://{{host}}:{{ports[http][0]}}/content-security-policy/support/");
+  url.hostname = host == Host.SAME_ORIGIN ? "{{host}}" : "{{domains[天気の良い日]}}";
+  url.pathname += path;
+
+  return url.toString();
+}
+
+function generateRedirect(host, target) {
+  var url = "http://{{host}}:{{ports[http][0]}}/common/redirect.py?location=" + encodeURIComponent(target);
+  if (host == Host.CROSS_ORIGIN) 
+    url = "http://{{domains[天気の良い日]}}:{{ports[http][0]}}/common/redirect.py?location=" + encodeURIComponent(target);
+
+  return url.toString();
+}
+
+function assert_embedding_csp(t, url, csp, expected, newCsp=null, newSrc=null) {
+  var i = document.createElement('iframe');
+  if (csp) 
+    i.csp = csp;
+  i.src = url;
+
+  window.addEventListener('message', t.step_func(e => {
+    if (e.source != i.contentWindow || !('embedding_csp' in e.data))
+        return;
+    if (newSrc && newCsp)
+      expected = newCsp;
+    assert_equals(expected, e.data['embedding_csp']);
+    t.done();
+  }));
+
+  document.body.appendChild(i);
+  if (newCsp)
+    i.csp = newCsp;
+  if (newSrc)
+    i.src = newSrc;
```

Comment:
I added it as a separate test to make sure the iframe has loaded and then changing `src` attribute to cross origin url.